block/manager.go

```diff
@@ -374,6 +410,39 @@ func (m *Manager) getRemainingSleep(start time.Time) time.Duration {
 	return 0
 }
 
+// BatchRetrieveLoop is responsible for retrieving batches from the sequencer.
+func (m *Manager) BatchRetrieveLoop(ctx context.Context) {
+	// batchTimer is used to signal when to retrieve batch from the sequencer
+	batchTimer := time.NewTimer(0)
+	defer batchTimer.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-batchTimer.C:
+			// Define the start time for the block production period
+			start := time.Now()
+			batch, err := m.seqClient.GetNextBatch(ctx, m.lastBatch)
+			if err != nil && ctx.Err() == nil {
+				m.logger.Error("error while retrieving batch", "error", err)
+			}
+			// Add the batch to the batch queue
+			if batch != nil && batch.Transactions != nil {
+				m.bq.AddBatch(*batch)
+				m.lastBatch = batch
+			}
+			// Reset the batchTimer to signal the next batch production
+			// period based on the batch retrieval time.
+			remainingSleep := time.Duration(0)
+			elapsed := time.Since(start)
+			if elapsed < defaultBatchRetrievalInterval {
+				remainingSleep = defaultBatchRetrievalInterval - elapsed
+			}
```

Comment:
In the comment for `defaultBatchRetrievalInterval` it says it is only used if the `DABlockTime` isn't set. 
This looks like it is always used. 

Should the comment be updated? Or how are the two variables related?

Reply:
it was a copied over comment. fixed it.